client-side/src/components/Report/report.jsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import PdfGenerator from './pdf.jsx';
+import { useState } from 'react';
+import Select from '../../stories/Select/Select.jsx';
+
+export default function Report() {
+  const [selectType, setSelectType] = useState("month");
+  const selectFunction=(selectedValue)=>{
+    if(selectedValue==1){
```

Comment:
please use enum that has key and value instead of all those it else,
something like:
in constants file should be:
const DATES_SELECT_OPTIONS = {
   DAY: {value: 1, text: 'day', iconSrc:'/images/time-twenty-four.svg'}
    ....
};
herein the component use it something like:
const setSelectedOption = selectedValue => {
 
 
 setSelectedType(DATES_SELECT_OPTIONS[selectedValue.key].value)
}

Reply:
you resolved only part of it, you added enum, but you didn't remove the if's